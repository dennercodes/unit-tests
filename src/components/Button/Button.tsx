interface ButtonProps {
  onClick?: () => void;
  className?: string;
  children: React.ReactNode;
}

export const Button = ({ onClick, children, className }: ButtonProps) => {
  return (
    <button onClick={onClick} className={className}>
      {children}
    </button>
  );
};
