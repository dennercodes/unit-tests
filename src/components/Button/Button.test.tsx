import { describe, it, expect, vi } from 'vitest';
import { render, screen } from '@testing-library/react';
import { Button } from './Button';
import userEvent from '@testing-library/user-event';

describe('Button', () => {
  it('should render default button', () => {
    // given
    render(<Button className="bg-red-500">Default button</Button>);
    // when
    const button = screen.getByRole('button', { name: /Default button/i });
    // then
    expect(button).matchSnapshot();
  });

  it('should render button with custom className', () => {
    // given
    render(<Button className="bg-red-500">Custom button</Button>);
    // when
    const button = screen.getByRole('button', { name: /Custom button/i });
    // then
    expect(button).toHaveClass('bg-red-500');
  });

  it('should render button', () => {
    // given
    render(<Button>I'm a button</Button>);
    // when
    const button = screen.getByRole('button', { name: /I'm a button/i });
    // then
    expect(button).toBeInTheDocument();
  });

  it('should call onClick when clicked', async () => {
    // given
    const onClick = vi.fn();
    render(<Button onClick={onClick}>I'm a button</Button>);
    // when
    const button = screen.getByRole('button', { name: /I'm a button/i });
    // then
    expect(button).toBeInTheDocument();
    // when
    await userEvent.click(button);
    // then
    expect(onClick).toHaveBeenCalled();
  });
});
