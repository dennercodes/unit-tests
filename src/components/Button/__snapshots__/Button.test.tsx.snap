// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > should render default button 1`] = `
<button
  class="bg-red-500"
>
  Default button
</button>
`;
